File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        hashmap={}
        left=0
        m=0
        for right in range(0,len(s)):
            if s[right] in hashmap and hashmap[s[right]]>=left:
                left=hashmap[s[right]]+1
            hashmap[s[right]]=right
            m=max(m,right-left+1)
        return m
            
        

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) 
    {
        List<Integer> r = new ArrayList<>();
        Stack<TreeNode> s = new Stack<>();
        TreeNode cur = root;
        while(k>0 && (cur!=null||!s.isEmpty()))
        {
            while(cur!=null)
            {
                s.push(cur);
                cur=cur.left;
            }
            cur = s.pop();
            r.add(cur.val);
            cur=cur.right;
            k--;
        }
        return r.getLast();
    }
}
